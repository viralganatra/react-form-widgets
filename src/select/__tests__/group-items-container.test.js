import React from 'react';
import GroupItemsContainerHOC from '../group-items-container';

const Item = (props) => <div {...props}>Test</div>;

const props = {
    test: '1',
    shouldDisplay: true,
};

describe('GroupItemsContainer', () => {
    let Component;

    beforeAll(() => {
        Component = GroupItemsContainerHOC(Item);
    });

    it('should render correctly', () => {
        const wrapperNotDisplay = shallow(
            <Component {...props} shouldDisplay={false} />,
        );

        const wrapperDisplay = shallow(<Component {...props} />);

        expect(wrapperNotDisplay).toMatchSnapshot();
        expect(wrapperDisplay).toMatchSnapshot();
    });

    it('should validate the prop types', () => {
        const wrapperInvalid = () => shallow(
            <Component {...props} shouldDisplay={null} />,
        );

        expect(wrapperInvalid).toThrowErrorMatchingSnapshot();
    });
});
