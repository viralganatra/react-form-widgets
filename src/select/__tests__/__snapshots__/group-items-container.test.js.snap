exports[`GroupItemsContainer should render correctly 1`] = `null`;

exports[`GroupItemsContainer should render correctly 2`] = `
<div
  style={
    Object {
      "height": "300px",
      "overflow": "auto",
      "width": "220px",
    }
  }>
  <Item
    test="1" />
</div>
`;

exports[`GroupItemsContainer should validate the prop types 1`] = `
"Warning: Failed prop type: The prop \`shouldDisplay\` is marked as required in \`SelectGroupItemsContainer\`, but its value is \`null\`.
    in SelectGroupItemsContainer"
`;
